samples/bluetooth/peripheral_hids_mouse/src/main.c

```diff
@@ -86,6 +86,14 @@ K_MSGQ_DEFINE(hids_queue,
 	      HIDS_QUEUE_SIZE,
 	      4);
 
+/* Bonded address queue. */
```

Comment:
Could compile out this queue and the adv_work when DIRECTED is not enabled.

Reply:
Okay, but this change is not that simple. I will need to add more #ifdefs to make sure that the compiled out queue is not accessed anywhere else.

I didn't use #ifdefs at first, because they make the code look somewhat obscure and it seemed to me that the sample clarity is more important than memory optimization. But again, this is debatable.